datafusion/common/src/scalar.rs

```diff
@@ -744,13 +744,13 @@ impl ScalarValue {
     pub fn new_primitive<T: ArrowPrimitiveType>(
         a: Option<T::Native>,
         d: &DataType,
-    ) -> Self {
+    ) -> Result<Self> {
```

Comment:
I think the comments in this function also need to be updated (the 'panic's section is no longer correct)

```
    /// Create a [`ScalarValue`] with the provided value and datatype
    ///
    /// # Panics
    ///
    /// Panics if d is not compatible with T
```

Reply:
with_data_type can still panic

Reply:
Wrong with what? A panic?

Reply:
Its an exceptional case that would imply some sort of schema inconsistency within the running plan, so imo panicking is perfectly valid, but then so is returning an error. The broader point is the doc is still correct

Reply:
I understand the point about the panics -- thank you

I don't have a strong opinion either way with regards to propagating errors vs panic'ing

Reply:
@alamb, I rewrote the comment and left a mention about panic. As @tustvold metioned `with_data_type` can panic.